### ISSUE TITLE
WMMA convolution kernel crashes on 7900 XT

### ISSUE THREAD

**Issue by atafra** [ROCmSoftwarePlatform/composable_kernel#634]
I tried to use the WMMA convolution kernel on a Radeon 7900 XT, ROCm 5.4.3 but it crashes due to an invalid memory access. The issue is not in my own code because the provided examples do not work either, and it's not just convolution. I tried commits 5b57ab9, 76fcdc6, 0ccecc7 and 9096b1c but neither of them work, the example and my own code too crash in all cases.

Here's the output when running a convolution example (no arguments passed):
```
./example_grouped_conv_fwd_bias_relu_add_wmma_fp16
in: dim 5, lengths {32, 2, 192, 71, 71}, strides {192, 30971904, 1, 436224, 6144}
wei: dim 5, lengths {32, 256, 192, 3, 3}, strides {442368, 1728, 1, 576, 192}
bias: dim 5, lengths {32, 2, 256, 36, 36}, strides {256, 0, 1, 0, 0}
residual: dim 5, lengths {32, 2, 256, 36, 36}, strides {256, 0, 1, 0, 0}
out: dim 5, lengths {32, 2, 256, 36, 36}, strides {256, 10616832, 1, 294912, 8192}
Memory access fault by GPU node-1 (Agent handle: 0x1890220) on address 0x7fb74bf08000. Reason: Page not present or supervisor privilege.
Aborted (core dumped)
```

The GEMM kernel produces incorrect output with the latest 5b57ab9, but it seems to work correctly using older commits:

```
./example_batched_gemm_bias_e_permute_wmma_fp16
a_gs_ms_ks: dim 5, lengths {1, 2, 4, 128, 2048}, strides {2097152, 1048576, 262144, 2048, 1}
b_gs_ns_ks: dim 5, lengths {1, 2, 16, 256, 2048}, strides {16777216, 8388608, 524288, 2048, 1}
d_gs_ms_ns: dim 6, lengths {1, 2, 4, 128, 16, 256}, strides {8192, 4096, 0, 0, 256, 1}
e_gs_ms_ns: dim 6, lengths {1, 2, 4, 128, 16, 256}, strides {4194304, 2097152, 524288, 256, 32768, 1}
GMNK=2, 512, 4096, 2048
Perf: 0.197592 ms, 86.9461 TFlops, 275.952 GB/s, DeviceBatchedContractionMultipleD_Wmma_CShuffle<256, 128, 256, 8, 8, 16, 16, 4, 4> NumPrefetch: 1, LoopScheduler: Default, PipelineVersion: v1
Error: Incorrect results!        out[0] != ref[0]: -1 != 842
Error: Incorrect results!        out[1] != ref[1]: -4 != 652
Error: Incorrect results!        out[2] != ref[2]: -1 != 122
Error: Incorrect results!        out[3] != ref[3]: 4 != 609
   max err: 2431
```

What could be the issue? I set `GPU_TARGETS=gfx1100` when building, and I checked that `hipcc` was indeed invoked with the correct architecture flag.

**Comment by aska-0096** [comment]
Hi @atafra Thanks for the bug report.
Let me clarify bugs you report to two type: 1. Memory access fault. 2. Incorrect result.

- Second issue caused by unsuitable arch limitation added to low-level intrinsic function call in the last commit.

https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/5b57ab96a8208eec1969a3dcadb555a6246ddb95/include/ck/utility/amd_wmma.hpp#L28-L31
We try to using a abbreviation to cover all gfx110x family but it fails, I will replace it with gfx1100 as a hotfix and improve it in near future.

- First issue also solved by doing above replacement, but it also could be result from bad GPU state, try call:

`rocm-smi -d 0 --gpureset`
This command help you reset your 7900XT, if you work inside docker, please call this command outside docker to make sure it effect.

**Comment by aska-0096** [comment]
PR created #639 .
You can also reach out to me if have further questions about Navi3x related operations.

**Comment by atafra** [comment]
Hi @aska-0096 . Thanks for the quick response but unfortunately this patch did not fix the crash in the convolution kernel for me. I tried to reset the GPU too but it didn't help either (I'm not using Docker). However, it did fix the incorrect output for GEMM. Here's the output:

```
rocm-smi -d 0 --gpureset
======================= ROCm System Management Interface =======================
================================== Reset GPU ===================================
GPU[0]          : Successfully reset GPU 0
================================================================================

./example_grouped_conv_fwd_bias_relu_add_wmma_fp16
in: dim 5, lengths {32, 2, 192, 71, 71}, strides {192, 30971904, 1, 436224, 6144}
wei: dim 5, lengths {32, 256, 192, 3, 3}, strides {442368, 1728, 1, 576, 192}
bias: dim 5, lengths {32, 2, 256, 36, 36}, strides {256, 0, 1, 0, 0}
residual: dim 5, lengths {32, 2, 256, 36, 36}, strides {256, 0, 1, 0, 0}
out: dim 5, lengths {32, 2, 256, 36, 36}, strides {256, 10616832, 1, 294912, 8192}
Memory access fault by GPU node-1 (Agent handle: 0x22fb220) on address 0x7ff6f3eea000. Reason: Page not present or supervisor privilege.
Aborted (core dumped)
```

So you cannot reproduce this crash in `example_grouped_conv_fwd_bias_relu_add_wmma_fp16` with the patch?

**Comment by aska-0096** [comment]
Yes, I cannot reproduce your memory access fault on this PR branch
My test environment is slightly different with you,

- Board: 7900 XTX, engineering example
- Runtime: ROCm 5.4.1
- Compiler: amd-stg-open

I would recommend you using a updated compiler, here is the procedure:
1. git pull -b amd-stg-open https://github.com/RadeonOpenCompute/llvm-project.git
2. cd llvm-project && mkdir -p build && cd build
3. cmake -DCMAKE_BUILD_TYPE=Release -DLLVM_ENABLE_ASSERTIONS=1 -DLLVM_TARGETS_TO_BUILD="AMDGPU;X86" -DLLVM_ENABLE_PROJECTS="clang;lld;compiler-rt" ../llvm
4. make -j install
5. export HIP_CLANG_PATH=/dockerx/llvm-project/bin/

Here is the cmake I used to build Navi3x kernels
```
-D CMAKE_PREFIX_PATH=/opt/rocm/                                           \
-D CMAKE_CXX_COMPILER=/opt/rocm/bin/hipcc                                \
-D CMAKE_CXX_FLAGS="-std=c++17 -O3 -ftemplate-backtrace-limit=0 -fPIE -mcumode -mno-wavefrontsize64 -Wno-gnu-line-marker -save-temps=$PWD"       \
-D CMAKE_BUILD_TYPE=Release                                                                       \
-D GPU_TARGETS="gfx1100"                                                                          \
..
```

**Comment by atafra** [comment]
I followed your instructions and built the amd-stg-open compiler but I still got the same crash. Maybe it crashes because it's an XT card? I double-checked that I'm indeed using the newer compiler:

```
hipcc -v
clang version 17.0.0 (https://github.com/RadeonOpenCompute/llvm-project.git c361f03b841f907bf2b6bf38f1e0baedeab4d051)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/aafra/llvm-install/bin
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/11
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/12
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/12
Candidate multilib: .;@m64
Selected multilib: .;@m64
Found HIP installation: /opt/rocm, version 5.4.22804
```

**Comment by aska-0096** [comment]
Good, at least we make sure the compiler is not the root cause.
I would like to recommend you run a small example with:
`./bin/example_grouped_conv_fwd_bias_relu_add_wmma_fp16 0 1 1 2 1 2 32 32 3 3 13 13 1 1 1 1 1 1 1 1
`
This need only very small amout of memory, may help us narrow the bug scope to memory allocation size.

If this example not works, please try following docker, this docker have exact HIP runtime as my environment, also do not forget use latest compiler inside docker.
` rocm/pytorch:rocm5.4.1_ubuntu20.04_py3.7_pytorch_1.12.1
`

If Compiler is same, Runtime is same, but code still not work, the only possible reason maybe the board, but unfortunately, I don't have a 7900XT card to reproduce. :(

Finally, if you allow me access your server, you can send me related information to mail address shown in my profile so that I can know what happened on 7900XT

**Comment by atafra** [comment]
I ran `./example_grouped_conv_fwd_bias_relu_add_wmma_fp16 0 1 1 2 1 2 32 32 3 3 13 13 1 1 1 1 1 1 1 1` but instead of crashing it just hangs after printing the params. Seems to be some deadlock. I can kill the process easily.

Unfortunately I can't give you access to the server because it's not a personal machine.

I'll try that Docker image next. Could you perhaps share your `example_grouped_conv_fwd_bias_relu_add_wmma_fp16` binary? This way we could make sure we're really running the same thing, and it would be much easier for me to set up.

**Comment by atafra** [comment]
I ran your binaries in the Docker container you're using, and I still get the same crash. So I think we can rule out compiler and software stack issues. The only relevant difference between your setup and mine is that you're using an XTX engineering sample and I'm using an XT retail.

I think it's very likely that this issue could be reproduced on any 7900 XT, not just my machine. The corresponding GEMM kernel works correctly, so this suggests that there's some bug in the convolution kernel, which is somehow not triggered on an XTX.

Could you try to reproduce this on a 7900 XT? Is there anything else we could try?

**Comment by aska-0096** [comment]
Hi @atafra 
Got it. Let me find a 7900 XT machine and give you feedback.

**Comment by atafra** [comment]
Thanks!

**Comment by atafra** [comment]
@aska-0096 Do you have any updates? Thanks!

**Comment by aska-0096** [comment]
> @aska-0096 Do you have any updates? Thanks!

Hi @atafra . The latest thing I can tell is that I have not found the available server resource yet. 

cc. @zjing14

**Comment by zjing14** [comment]
@aska-0096 @atafra We are trying to get a 7900xt card internally. If not, we may just buy one.

**Comment by aska-0096** [comment]
Hi @atafra , a convolution related bug fixed, let me know is it work for you. check it out here #649

**Comment by zjing14** [comment]
@atafra Could you set env var `export AMD_LOG_LEVEL=4` and post logs here?

**Comment by atafra** [comment]
Here's the log:

```
./example_grouped_conv_fwd_bias_relu_add_wmma_fp16
in: dim 5, lengths {32, 2, 192, 71, 71}, strides {192, 30971904, 1, 436224, 6144}
wei: dim 5, lengths {32, 256, 192, 3, 3}, strides {442368, 1728, 1, 576, 192}
bias: dim 5, lengths {32, 2, 256, 36, 36}, strides {256, 0, 1, 0, 0}
residual: dim 5, lengths {32, 2, 256, 36, 36}, strides {256, 0, 1, 0, 0}
out: dim 5, lengths {32, 2, 256, 36, 36}, strides {256, 10616832, 1, 294912, 8192}
:3:rocdevice.cpp            :426 : 15459317583 us: 231610: [tid:0x7f29846703c0] Initializing HSA stack.
:3:comgrctx.cpp             :33  : 15459324463 us: 231610: [tid:0x7f29846703c0] Loading COMGR library.
:3:rocdevice.cpp            :205 : 15459327250 us: 231610: [tid:0x7f29846703c0] Numa selects cpu agent[0]=0x1d5a860(fine=0x1d5aa50,coarse=0x1d5af10) for gpu agent=0x1d584f0
:3:rocdevice.cpp            :1633: 15459327390 us: 231610: [tid:0x7f29846703c0] HMM support: 1, xnack: 0, direct host access: 0

:4:rocdevice.cpp            :1942: 15459327414 us: 231610: [tid:0x7f29846703c0] Allocate hsa host memory 0x7f29847ea000, size 0x28
:4:rocdevice.cpp            :1942: 15459327556 us: 231610: [tid:0x7f29846703c0] Allocate hsa host memory 0x7f2972c00000, size 0x101000
:4:rocdevice.cpp            :1942: 15459327686 us: 231610: [tid:0x7f29846703c0] Allocate hsa host memory 0x7f2972a00000, size 0x101000
:4:runtime.cpp              :83  : 15459327697 us: 231610: [tid:0x7f29846703c0] init
:3:hip_context.cpp          :50  : 15459327699 us: 231610: [tid:0x7f29846703c0] Direct Dispatch: 1
:4:rocdevice.cpp            :2069: 15459327741 us: 231610: [tid:0x7f29846703c0] Allocate hsa device memory 0x7f2854600000, size 0x7626000
:3:rocdevice.cpp            :2108: 15459327743 us: 231610: [tid:0x7f29846703c0] device=0x1d6b970, freeMem_ = 0xf79da000
:3:hip_memory.cpp           :513 : 15459327751 us: 231610: [tid:0x7f29846703c0] hipMalloc: Returned hipSuccess : 0x7f2854600000
:3:hip_memory.cpp           :511 : 15459327756 us: 231610: [tid:0x7f29846703c0]  hipMalloc ( 0x7ffcada254c8, 28311552 )
:4:rocdevice.cpp            :2069: 15459327764 us: 231610: [tid:0x7f29846703c0] Allocate hsa device memory 0x7f2970c00000, size 0x1b00000
:3:rocdevice.cpp            :2108: 15459327766 us: 231610: [tid:0x7f29846703c0] device=0x1d6b970, freeMem_ = 0xf5eda000
:3:hip_memory.cpp           :513 : 15459327767 us: 231610: [tid:0x7f29846703c0] hipMalloc: Returned hipSuccess : 0x7f2970c00000: duration: 11 us
:3:hip_memory.cpp           :511 : 15459327770 us: 231610: [tid:0x7f29846703c0]  hipMalloc ( 0x7ffcada254b8, 16384 )
:4:rocdevice.cpp            :2069: 15459327777 us: 231610: [tid:0x7f29846703c0] Allocate hsa device memory 0x7f2970800000, size 0x4000
:3:rocdevice.cpp            :2108: 15459327779 us: 231610: [tid:0x7f29846703c0] device=0x1d6b970, freeMem_ = 0xf5ed6000
:3:hip_memory.cpp           :513 : 15459327781 us: 231610: [tid:0x7f29846703c0] hipMalloc: Returned hipSuccess : 0x7f2970800000: duration: 11 us
:3:hip_memory.cpp           :511 : 15459327783 us: 231610: [tid:0x7f29846703c0]  hipMalloc ( 0x7ffcada254a8, 16384 )
:4:rocdevice.cpp            :2069: 15459327786 us: 231610: [tid:0x7f29846703c0] Allocate hsa device memory 0x7f2970804000, size 0x4000
:3:rocdevice.cpp            :2108: 15459327788 us: 231610: [tid:0x7f29846703c0] device=0x1d6b970, freeMem_ = 0xf5ed2000
:3:hip_memory.cpp           :513 : 15459327789 us: 231610: [tid:0x7f29846703c0] hipMalloc: Returned hipSuccess : 0x7f2970804000: duration: 6 us
:3:hip_memory.cpp           :511 : 15459327791 us: 231610: [tid:0x7f29846703c0]  hipMalloc ( 0x7ffcada25498, 42467328 )
:4:rocdevice.cpp            :2069: 15459327799 us: 231610: [tid:0x7f29846703c0] Allocate hsa device memory 0x7f2861000000, size 0x2880000
:3:rocdevice.cpp            :2108: 15459327800 us: 231610: [tid:0x7f29846703c0] device=0x1d6b970, freeMem_ = 0xf3652000
:3:hip_memory.cpp           :513 : 15459327802 us: 231610: [tid:0x7f29846703c0] hipMalloc: Returned hipSuccess : 0x7f2861000000: duration: 11 us
:3:hip_memory.cpp           :582 : 15459327808 us: 231610: [tid:0x7f29846703c0]  hipMemcpy ( 0x7f2854600000, 0x7f297a9d9010, 123887616, hipMemcpyHostToDevice )
:3:rocdevice.cpp            :2703: 15459327816 us: 231610: [tid:0x7f29846703c0] number of allocated hardware queues with low priority: 0, with normal priority: 0, with high priority: 0, maximum per priority is: 4
:3:rocdevice.cpp            :2777: 15459334025 us: 231610: [tid:0x7f29846703c0] created hardware queue 0x7f2984600000 with size 16384 with priority 1, cooperative: 0
:4:rocdevice.cpp            :1942: 15459334159 us: 231610: [tid:0x7f29846703c0] Allocate hsa host memory 0x7f2970400000, size 0x100000
:3:devprogram.cpp           :2676: 15459441247 us: 231610: [tid:0x7f29846703c0] Using Code Object V4.
:3:devprogram.cpp           :2979: 15459441935 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: __amd_rocclr_streamOpsWrite
:3:devprogram.cpp           :2979: 15459441939 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: __amd_rocclr_fillImage
:3:devprogram.cpp           :2979: 15459441940 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: __amd_rocclr_fillBufferAligned2D
:3:devprogram.cpp           :2979: 15459441941 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: __amd_rocclr_copyImageToBuffer
:3:devprogram.cpp           :2979: 15459441942 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: __amd_rocclr_copyImage1DA
:3:devprogram.cpp           :2979: 15459441943 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: __amd_rocclr_fillBufferAligned
:3:devprogram.cpp           :2979: 15459441944 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: __amd_rocclr_copyImage
:3:devprogram.cpp           :2979: 15459441945 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: __amd_rocclr_streamOpsWait
:3:devprogram.cpp           :2979: 15459441946 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: __amd_rocclr_copyBufferRectAligned
:3:devprogram.cpp           :2979: 15459441947 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: __amd_rocclr_copyBufferRect
:3:devprogram.cpp           :2979: 15459441948 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: __amd_rocclr_copyBufferAligned
:3:devprogram.cpp           :2979: 15459441949 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: __amd_rocclr_copyBufferToImage
:3:devprogram.cpp           :2979: 15459441950 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: __amd_rocclr_copyBuffer
:4:command.cpp              :349 : 15459441981 us: 231610: [tid:0x7f29846703c0] Command (CopyHostToDevice) enqueued: 0x22aa740
:4:rocmemory.cpp            :944 : 15459442562 us: 231610: [tid:0x7f29846703c0] Locking to pool 0x1d5af10, size 0x2001000, HostPtr = 0x7f297a9d9000, DevPtr = 0x7f284e800000
:4:rocblit.cpp              :664 : 15459442567 us: 231610: [tid:0x7f29846703c0] HSA Asycn Copy dst=0x7f2854600000, src=0x7f284e800010, size=0, wait_event=0x7f29847ee800, completion_signal=0x0
:4:rocvirtual.cpp           :550 : 15459443487 us: 231610: [tid:0x7f29846703c0] Host wait on completion_signal=0x7f29847ee800
:3:rocvirtual.hpp           :61  : 15459443489 us: 231610: [tid:0x7f29846703c0] Host active wait for Signal = (0x7f29847ee800) for -1 ns
:4:rocmemory.cpp            :944 : 15459445686 us: 231610: [tid:0x7f29846703c0] Locking to pool 0x1d5af10, size 0x2000000, HostPtr = 0x7f297c9da000, DevPtr = 0x7f284e800000
:4:rocblit.cpp              :664 : 15459445688 us: 231610: [tid:0x7f29846703c0] HSA Asycn Copy dst=0x7f2856600ff0, src=0x7f284e800000, size=0, wait_event=0x7f29847ee780, completion_signal=0x0
:4:rocvirtual.cpp           :550 : 15459445690 us: 231610: [tid:0x7f29846703c0] Host wait on completion_signal=0x7f29847ee780
:3:rocvirtual.hpp           :61  : 15459445691 us: 231610: [tid:0x7f29846703c0] Host active wait for Signal = (0x7f29847ee780) for -1 ns
:4:rocmemory.cpp            :944 : 15459447519 us: 231610: [tid:0x7f29846703c0] Locking to pool 0x1d5af10, size 0x2000000, HostPtr = 0x7f297e9da000, DevPtr = 0x7f284e800000
:4:rocblit.cpp              :664 : 15459447521 us: 231610: [tid:0x7f29846703c0] HSA Asycn Copy dst=0x7f2858600ff0, src=0x7f284e800000, size=0, wait_event=0x7f29847ee700, completion_signal=0x0
:4:rocvirtual.cpp           :550 : 15459447522 us: 231610: [tid:0x7f29846703c0] Host wait on completion_signal=0x7f29847ee700
:3:rocvirtual.hpp           :61  : 15459447523 us: 231610: [tid:0x7f29846703c0] Host active wait for Signal = (0x7f29847ee700) for -1 ns
:4:rocmemory.cpp            :944 : 15459449049 us: 231610: [tid:0x7f29846703c0] Locking to pool 0x1d5af10, size 0x1626000, HostPtr = 0x7f29809da000, DevPtr = 0x7f284f200000
:4:rocblit.cpp              :664 : 15459449050 us: 231610: [tid:0x7f29846703c0] HSA Asycn Copy dst=0x7f285a600ff0, src=0x7f284f200000, size=0, wait_event=0x7f29847ee680, completion_signal=0x0
:4:rocvirtual.cpp           :550 : 15459449052 us: 231610: [tid:0x7f29846703c0] Host wait on completion_signal=0x7f29847ee680
:3:rocvirtual.hpp           :61  : 15459449053 us: 231610: [tid:0x7f29846703c0] Host active wait for Signal = (0x7f29847ee680) for -1 ns
:4:command.cpp              :289 : 15459449908 us: 231610: [tid:0x7f29846703c0] Queue marker to command queue: 0x1d48940:4:command.cpp              :349 : 15459449909 us: 231610: [tid:0x7f29846703c0] Command (InternalMarker) enqueued: 0x21171a0
:4:command.cpp              :179 : 15459449911 us: 231610: [tid:0x7f29846703c0] Command 0x22aa740 complete
:4:command.cpp              :173 : 15459449912 us: 231610: [tid:0x7f29846703c0] Command 0x21171a0 complete (Wall: 15459449912, CPU: 0, GPU: 0 us)
:4:command.cpp              :253 : 15459449914 us: 231610: [tid:0x7f29846703c0] Waiting for event 0x22aa740 to complete, current status 0
:4:command.cpp              :268 : 15459449915 us: 231610: [tid:0x7f29846703c0] Event 0x22aa740 wait completed
:3:hip_memory.cpp           :583 : 15459449917 us: 231610: [tid:0x7f29846703c0] hipMemcpy: Returned hipSuccess : : duration: 122109 us
:3:hip_memory.cpp           :582 : 15459449921 us: 231610: [tid:0x7f29846703c0]  hipMemcpy ( 0x7f2970c00000, 0x7f2978ed8010, 28311552, hipMemcpyHostToDevice )
:4:command.cpp              :349 : 15459449924 us: 231610: [tid:0x7f29846703c0] Command (CopyHostToDevice) enqueued: 0x22aa740
:4:rocmemory.cpp            :944 : 15459450199 us: 231610: [tid:0x7f29846703c0] Locking to pool 0x1d5af10, size 0x1b01000, HostPtr = 0x7f2978ed8000, DevPtr = 0x7f284ee00000
:4:rocblit.cpp              :664 : 15459450201 us: 231610: [tid:0x7f29846703c0] HSA Asycn Copy dst=0x7f2970c00000, src=0x7f284ee00010, size=0, wait_event=0x7f29847ee600, completion_signal=0x0
:4:rocvirtual.cpp           :550 : 15459450202 us: 231610: [tid:0x7f29846703c0] Host wait on completion_signal=0x7f29847ee600
:3:rocvirtual.hpp           :61  : 15459450203 us: 231610: [tid:0x7f29846703c0] Host active wait for Signal = (0x7f29847ee600) for -1 ns
:4:command.cpp              :289 : 15459451240 us: 231610: [tid:0x7f29846703c0] Queue marker to command queue: 0x1d48940:4:command.cpp              :349 : 15459451241 us: 231610: [tid:0x7f29846703c0] Command (InternalMarker) enqueued: 0x21171a0
:4:command.cpp              :179 : 15459451243 us: 231610: [tid:0x7f29846703c0] Command 0x22aa740 complete
:4:command.cpp              :173 : 15459451244 us: 231610: [tid:0x7f29846703c0] Command 0x21171a0 complete (Wall: 15459451243, CPU: 0, GPU: 0 us)
:4:command.cpp              :253 : 15459451245 us: 231610: [tid:0x7f29846703c0] Waiting for event 0x22aa740 to complete, current status 0
:4:command.cpp              :268 : 15459451246 us: 231610: [tid:0x7f29846703c0] Event 0x22aa740 wait completed
:3:hip_memory.cpp           :583 : 15459451247 us: 231610: [tid:0x7f29846703c0] hipMemcpy: Returned hipSuccess : : duration: 1326 us
:3:hip_memory.cpp           :582 : 15459451250 us: 231610: [tid:0x7f29846703c0]  hipMemcpy ( 0x7f2970800000, 0x1d4b790, 16384, hipMemcpyHostToDevice )
:4:command.cpp              :349 : 15459451254 us: 231610: [tid:0x7f29846703c0] Command (CopyHostToDevice) enqueued: 0x22aa740
:4:rocblit.cpp              :724 : 15459451258 us: 231610: [tid:0x7f29846703c0] HSA Async Copy staged H2D dst=0x7f2970800000, src=0x7f2972c00000, size=16384, completion_signal=0x7f29847ee580
:4:rocvirtual.cpp           :550 : 15459451261 us: 231610: [tid:0x7f29846703c0] Host wait on completion_signal=0x7f29847ee580
:3:rocvirtual.hpp           :61  : 15459451262 us: 231610: [tid:0x7f29846703c0] Host active wait for Signal = (0x7f29847ee580) for -1 ns
:4:command.cpp              :289 : 15459451271 us: 231610: [tid:0x7f29846703c0] Queue marker to command queue: 0x1d48940:4:command.cpp              :349 : 15459451273 us: 231610: [tid:0x7f29846703c0] Command (InternalMarker) enqueued: 0x21171a0
:4:command.cpp              :179 : 15459451274 us: 231610: [tid:0x7f29846703c0] Command 0x22aa740 complete
:4:command.cpp              :173 : 15459451275 us: 231610: [tid:0x7f29846703c0] Command 0x21171a0 complete (Wall: 15459451275, CPU: 0, GPU: 0 us)
:4:command.cpp              :253 : 15459451276 us: 231610: [tid:0x7f29846703c0] Waiting for event 0x22aa740 to complete, current status 0
:4:command.cpp              :268 : 15459451277 us: 231610: [tid:0x7f29846703c0] Event 0x22aa740 wait completed
:3:hip_memory.cpp           :583 : 15459451280 us: 231610: [tid:0x7f29846703c0] hipMemcpy: Returned hipSuccess : : duration: 30 us
:3:hip_memory.cpp           :582 : 15459451281 us: 231610: [tid:0x7f29846703c0]  hipMemcpy ( 0x7f2970804000, 0x1d4f7a0, 16384, hipMemcpyHostToDevice )
:4:command.cpp              :349 : 15459451283 us: 231610: [tid:0x7f29846703c0] Command (CopyHostToDevice) enqueued: 0x22aa740
:4:rocblit.cpp              :724 : 15459451286 us: 231610: [tid:0x7f29846703c0] HSA Async Copy staged H2D dst=0x7f2970804000, src=0x7f2972c00000, size=16384, completion_signal=0x7f29847ee500
:4:rocvirtual.cpp           :550 : 15459451289 us: 231610: [tid:0x7f29846703c0] Host wait on completion_signal=0x7f29847ee500
:3:rocvirtual.hpp           :61  : 15459451290 us: 231610: [tid:0x7f29846703c0] Host active wait for Signal = (0x7f29847ee500) for -1 ns
:4:command.cpp              :289 : 15459451298 us: 231610: [tid:0x7f29846703c0] Queue marker to command queue: 0x1d48940:4:command.cpp              :349 : 15459451300 us: 231610: [tid:0x7f29846703c0] Command (InternalMarker) enqueued: 0x21171a0
:4:command.cpp              :179 : 15459451301 us: 231610: [tid:0x7f29846703c0] Command 0x22aa740 complete
:4:command.cpp              :173 : 15459451302 us: 231610: [tid:0x7f29846703c0] Command 0x21171a0 complete (Wall: 15459451302, CPU: 0, GPU: 0 us)
:4:command.cpp              :253 : 15459451303 us: 231610: [tid:0x7f29846703c0] Waiting for event 0x22aa740 to complete, current status 0
:4:command.cpp              :268 : 15459451306 us: 231610: [tid:0x7f29846703c0] Event 0x22aa740 wait completed
:3:hip_memory.cpp           :583 : 15459451307 us: 231610: [tid:0x7f29846703c0] hipMemcpy: Returned hipSuccess : : duration: 26 us
:3:hip_device_runtime.cpp   :526 : 15459451312 us: 231610: [tid:0x7f29846703c0]  hipGetDevice ( 0x7ffcada2489c )
:3:hip_device_runtime.cpp   :534 : 15459451314 us: 231610: [tid:0x7f29846703c0] hipGetDevice: Returned hipSuccess :
:3:hip_device.cpp           :346 : 15459451316 us: 231610: [tid:0x7f29846703c0]  hipGetDeviceProperties ( 0x7ffcada24c00, 0 )
:3:hip_device.cpp           :348 : 15459451319 us: 231610: [tid:0x7f29846703c0] hipGetDeviceProperties: Returned hipSuccess :
:3:hip_platform.cpp         :194 : 15459451331 us: 231610: [tid:0x7f29846703c0]  __hipPushCallConfiguration ( {1344,1,1}, {256,1,1}, 0, stream:<null> )
:3:hip_platform.cpp         :198 : 15459451334 us: 231610: [tid:0x7f29846703c0] __hipPushCallConfiguration: Returned hipSuccess :
:3:hip_platform.cpp         :203 : 15459451338 us: 231610: [tid:0x7f29846703c0]  __hipPopCallConfiguration ( {2913094120,32764,2913094116}, {2189611792,32553,2222973344}, 0x7ffcada24930, 0x7ffcada24928 )
:3:hip_platform.cpp         :212 : 15459451339 us: 231610: [tid:0x7f29846703c0] __hipPopCallConfiguration: Returned hipSuccess :
:3:hip_module.cpp           :469 : 15459455714 us: 231610: [tid:0x7f29846703c0]  hipLaunchKernel ( 0x207740, {1344,1,1}, {256,1,1}, 0x7ffcada24980, 0, stream:<null> )
:3:devprogram.cpp           :2676: 15459456021 us: 231610: [tid:0x7f29846703c0] Using Code Object V4.
:3:devprogram.cpp           :2979: 15459456532 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: _ZN2ck48kernel_grouped_conv_fwd_multiple_d_wmma_cshuffleINS_50GridwiseGemmMultipleD_k0mk1_k0nk1_mn_wmma_cshuffleIDF16_DF16_fDF16_NS_5TupleIJDF16_DF16_EEEDF16_NS_16TensorDescriptorINS2_IJNS_5EmbedINS2_IJiiiiiEEENS2_IJiiiiNS_17integral_constantIiLi1EEEEEELb0EEENS_11PassThroughIiEENS_3PadIiiiLb0EEESE_SE_SC_SC_NS5_INS2_IJiiEEESF_Lb0EEESG_SG_SC_NS_23Merge_v2_magic_divisionINS2_IJiiiiEEEEESJ_NS_8RightPadIiiLb0EEESL_NS_7UnMergeISF_Lb0EEESC_EEENS2_IJNS_8SequenceIJLi0EEEENSP_IJLi1EEEENSP_IJLi2EEEENSP_IJLi3EEEENSP_IJLi4EEEENSP_IJLi5EEEENSP_IJLi6EEEENSP_IJLi7EEEENSP_IJLi8EEEENSP_IJLi9EEEENSP_IJLi10EEEENSP_IJLi11ELi13ELi15ELi17EEEENSP_IJLi12ELi14ELi16ELi18EEEENSP_IJLi19EEEENSP_IJLi20EEEENSP_IJLi22EEEENSP_IJLi21EEEEEEENS2_IJNSP_IJLi1ELi2ELi3ELi4ELi5EEEESW_SX_SY_SZ_S10_NSP_IJLi11EEEENSP_IJLi12ELi13EEEENSP_IJLi14ELi15EEEENSP_IJLi16ELi17EEEENSP_IJLi18EEEES13_S14_S16_S15_NSP_IJLi23ELi24EEEENSP_IJLi25EEEEEEENSP_IJLi23ELi25ELi24EEEElEENS4_INS2_IJNS5_INS2_IJiiiEEENS2_IJiiS8_EEELb0EEESC_NSH_ISF_EESL_SL_SN_SC_EEENS2_IJSQ_SR_NSP_IJLi2ELi3EEEESU_SV_SX_SW_EEENS2_IJNSP_IJLi1ELi2ELi3EEEESU_SV_SW_SX_NSP_IJLi8ELi9EEEES10_EEENSP_IJLi8ELi10ELi9EEEElEENS2_IJNS4_INS2_IJNS5_ISF_NS2_IJNS7_IiLi0EEES8_EEELb0EEESL_SL_EEENS2_IJSQ_SR_SS_EEENS2_IJNSP_IJLi1ELi2EEEEST_SU_EEENSP_IJLi3ELi4EEEElEENS4_INS2_IJNS5_ISF_NS2_IJiS8_EEELb0EEESL_SL_EEES1Z_S21_S22_lEEEEES27_NS_16tensor_operation12element_wise11PassThroughES2B_NS2A_10AddReluAddELNS_25InMemoryDataOperationEnumE0ELi128ELi128ELi4ELi16ELi16ELi8ELi4ELi2ELi256ENSP_IJLi4ELi64ELi1EEEENSP_IJLi1ELi0ELi2EEEES2F_Li2ELi8ELi8ELb0ELb1ES2E_S2F_S2F_Li2ELi8ELi8ELb0ELb1ELi4ELi2ENSP_IJLi1ELi32ELi1ELi8EEEELi8ELi1ELNS_13LoopSchedulerE0ELNS_15PipelineVersionE0EEEDF16_DF16_NS2_IJPKDF16_S2L_EEEDF16_S2B_S2B_S2C_S1I_S1U_NS2_IJNS4_INS2_IJS1X_SL_SL_NSM_INS2_IJiNS7_IiLi128EEEEEELb0EEES2P_EEENS2_IJSQ_SR_SS_ST_SU_EEENS2_IJS20_ST_SU_NSP_IJLi5ELi6EEEENSP_IJLi7ELi8EEEEEEENSP_IJLi5ELi6ELi7ELi8EEEElEENS4_INS2_IJS25_SL_SL_S2P_S2P_EEES2R_S2U_S2V_lEEEEES2Y_NS_31BlockToCTileMap_M00_N0_M01AdaptILi128ELi128ES27_EENS29_6device12_GLOBAL__N_130ComputePtrOffsetOfStridedBatchILi2EEELb0EEEvPKT0_PKT1_T2_PT3_T4_T5_T6_iT7_T8_T9_T10_T11_T12_
:3:devprogram.cpp           :2979: 15459456536 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: _ZN2ck48kernel_grouped_conv_fwd_multiple_d_wmma_cshuffleINS_50GridwiseGemmMultipleD_k0mk1_k0nk1_mn_wmma_cshuffleIDF16_DF16_fDF16_NS_5TupleIJDF16_DF16_EEEDF16_NS_16TensorDescriptorINS2_IJNS_5EmbedINS2_IJiiiiEEENS2_IJiiiNS_17integral_constantIiLi1EEEEEELb0EEENS_11PassThroughIiEENS_3PadIiiiLb0EEESE_SC_SC_NS5_INS2_IJiiEEESF_Lb0EEESG_SC_NS_23Merge_v2_magic_divisionINS2_IJiiiEEEEESJ_NS_8RightPadIiiLb0EEESL_NS_7UnMergeISF_Lb0EEESC_EEENS2_IJNS_8SequenceIJLi0EEEENSP_IJLi1EEEENSP_IJLi2EEEENSP_IJLi3EEEENSP_IJLi4EEEENSP_IJLi5EEEENSP_IJLi6EEEENSP_IJLi7EEEENSP_IJLi8EEEENSP_IJLi9ELi11ELi13EEEENSP_IJLi10ELi12ELi14EEEENSP_IJLi15EEEENSP_IJLi16EEEENSP_IJLi18EEEENSP_IJLi17EEEEEEENS2_IJNSP_IJLi1ELi2ELi3ELi4EEEESV_SW_SX_SY_NSP_IJLi9EEEENSP_IJLi10ELi11EEEENSP_IJLi12ELi13EEEENSP_IJLi14EEEES11_S12_S14_S13_NSP_IJLi19ELi20EEEENSP_IJLi21EEEEEEENSP_IJLi19ELi21ELi20EEEElEENS4_INS2_IJNS5_ISI_NS2_IJiiS8_EEELb0EEESC_NSH_ISF_EESL_SL_SN_SC_EEENS2_IJSQ_SR_NSP_IJLi2ELi3EEEESU_SV_SX_SW_EEENS2_IJNSP_IJLi1ELi2ELi3EEEESU_SV_SW_SX_NSP_IJLi8ELi9EEEENSP_IJLi10EEEEEEENSP_IJLi8ELi10ELi9EEEElEENS2_IJNS4_INS2_IJNS5_ISF_NS2_IJNS7_IiLi0EEES8_EEELb0EEESL_SL_EEENS2_IJSQ_SR_SS_EEENS2_IJNSP_IJLi1ELi2EEEEST_SU_EEENSP_IJLi3ELi4EEEElEENS4_INS2_IJNS5_ISF_NS2_IJiS8_EEELb0EEESL_SL_EEES1W_S1Y_S1Z_lEEEEES24_NS_16tensor_operation12element_wise11PassThroughES28_NS27_10AddReluAddELNS_25InMemoryDataOperationEnumE0ELi128ELi128ELi4ELi16ELi16ELi8ELi4ELi2ELi256ENSP_IJLi4ELi64ELi1EEEENSP_IJLi1ELi0ELi2EEEES2C_Li2ELi8ELi8ELb0ELb1ES2B_S2C_S2C_Li2ELi8ELi8ELb0ELb1ELi4ELi2ENSP_IJLi1ELi32ELi1ELi8EEEELi8ELi1ELNS_13LoopSchedulerE0ELNS_15PipelineVersionE0EEEDF16_DF16_NS2_IJPKDF16_S2I_EEEDF16_S28_S28_S29_S1F_S1R_NS2_IJNS4_INS2_IJS1U_SL_SL_NSM_INS2_IJiNS7_IiLi128EEEEEELb0EEES2M_EEENS2_IJSQ_SR_SS_ST_SU_EEENS2_IJS1X_ST_SU_NSP_IJLi5ELi6EEEENSP_IJLi7ELi8EEEEEEENSP_IJLi5ELi6ELi7ELi8EEEElEENS4_INS2_IJS22_SL_SL_S2M_S2M_EEES2O_S2R_S2S_lEEEEES2V_NS_31BlockToCTileMap_M00_N0_M01AdaptILi128ELi128ES24_EENS26_6device12_GLOBAL__N_130ComputePtrOffsetOfStridedBatchILi2EEELb1EEEvPKT0_PKT1_T2_PT3_T4_T5_T6_iT7_T8_T9_T10_T11_T12_
:3:devprogram.cpp           :2979: 15459456540 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: _ZN2ck48kernel_grouped_conv_fwd_multiple_d_wmma_cshuffleINS_50GridwiseGemmMultipleD_k0mk1_k0nk1_mn_wmma_cshuffleIDF16_DF16_fDF16_NS_5TupleIJDF16_DF16_EEEDF16_NS_16TensorDescriptorINS2_IJNS_5EmbedINS2_IJiiiiEEENS2_IJiiiNS_17integral_constantIiLi1EEEEEELb0EEENS_11PassThroughIiEENS_3PadIiiiLb0EEESE_SC_SC_NS5_INS2_IJiiEEESF_Lb0EEESG_SC_NS_23Merge_v2_magic_divisionINS2_IJiiiEEEEESJ_NS_8RightPadIiiLb0EEESL_NS_7UnMergeISF_Lb0EEESC_EEENS2_IJNS_8SequenceIJLi0EEEENSP_IJLi1EEEENSP_IJLi2EEEENSP_IJLi3EEEENSP_IJLi4EEEENSP_IJLi5EEEENSP_IJLi6EEEENSP_IJLi7EEEENSP_IJLi8EEEENSP_IJLi9ELi11ELi13EEEENSP_IJLi10ELi12ELi14EEEENSP_IJLi15EEEENSP_IJLi16EEEENSP_IJLi18EEEENSP_IJLi17EEEEEEENS2_IJNSP_IJLi1ELi2ELi3ELi4EEEESV_SW_SX_SY_NSP_IJLi9EEEENSP_IJLi10ELi11EEEENSP_IJLi12ELi13EEEENSP_IJLi14EEEES11_S12_S14_S13_NSP_IJLi19ELi20EEEENSP_IJLi21EEEEEEENSP_IJLi19ELi21ELi20EEEElEENS4_INS2_IJNS5_ISI_NS2_IJiiS8_EEELb0EEESC_NSH_ISF_EESL_SL_SN_SC_EEENS2_IJSQ_SR_NSP_IJLi2ELi3EEEESU_SV_SX_SW_EEENS2_IJNSP_IJLi1ELi2ELi3EEEESU_SV_SW_SX_NSP_IJLi8ELi9EEEENSP_IJLi10EEEEEEENSP_IJLi8ELi10ELi9EEEElEENS2_IJNS4_INS2_IJNS5_ISF_NS2_IJNS7_IiLi0EEES8_EEELb0EEESL_SL_EEENS2_IJSQ_SR_SS_EEENS2_IJNSP_IJLi1ELi2EEEEST_SU_EEENSP_IJLi3ELi4EEEElEENS4_INS2_IJNS5_ISF_NS2_IJiS8_EEELb0EEESL_SL_EEES1W_S1Y_S1Z_lEEEEES24_NS_16tensor_operation12element_wise11PassThroughES28_NS27_10AddReluAddELNS_25InMemoryDataOperationEnumE0ELi128ELi128ELi4ELi16ELi16ELi8ELi4ELi2ELi256ENSP_IJLi4ELi64ELi1EEEENSP_IJLi1ELi0ELi2EEEES2C_Li2ELi8ELi8ELb0ELb1ES2B_S2C_S2C_Li2ELi8ELi8ELb0ELb1ELi4ELi2ENSP_IJLi1ELi32ELi1ELi8EEEELi8ELi1ELNS_13LoopSchedulerE0ELNS_15PipelineVersionE0EEEDF16_DF16_NS2_IJPKDF16_S2I_EEEDF16_S28_S28_S29_S1F_S1R_NS2_IJNS4_INS2_IJS1U_SL_SL_NSM_INS2_IJiNS7_IiLi128EEEEEELb0EEES2M_EEENS2_IJSQ_SR_SS_ST_SU_EEENS2_IJS1X_ST_SU_NSP_IJLi5ELi6EEEENSP_IJLi7ELi8EEEEEEENSP_IJLi5ELi6ELi7ELi8EEEElEENS4_INS2_IJS22_SL_SL_S2M_S2M_EEES2O_S2R_S2S_lEEEEES2V_NS_31BlockToCTileMap_M00_N0_M01AdaptILi128ELi128ES24_EENS26_6device12_GLOBAL__N_130ComputePtrOffsetOfStridedBatchILi2EEELb0EEEvPKT0_PKT1_T2_PT3_T4_T5_T6_iT7_T8_T9_T10_T11_T12_
:3:devprogram.cpp           :2979: 15459456543 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: _ZN2ck48kernel_grouped_conv_fwd_multiple_d_wmma_cshuffleINS_50GridwiseGemmMultipleD_k0mk1_k0nk1_mn_wmma_cshuffleIDF16_DF16_fDF16_NS_5TupleIJDF16_DF16_EEEDF16_NS_16TensorDescriptorINS2_IJNS_5EmbedINS2_IJiiiiiEEENS2_IJiiiiNS_17integral_constantIiLi1EEEEEELb0EEENS_11PassThroughIiEENS_3PadIiiiLb0EEESE_SE_SC_SC_NS5_INS2_IJiiEEESF_Lb0EEESG_SG_SC_NS_23Merge_v2_magic_divisionINS2_IJiiiiEEEEESJ_NS_8RightPadIiiLb0EEESL_NS_7UnMergeISF_Lb0EEESC_EEENS2_IJNS_8SequenceIJLi0EEEENSP_IJLi1EEEENSP_IJLi2EEEENSP_IJLi3EEEENSP_IJLi4EEEENSP_IJLi5EEEENSP_IJLi6EEEENSP_IJLi7EEEENSP_IJLi8EEEENSP_IJLi9EEEENSP_IJLi10EEEENSP_IJLi11ELi13ELi15ELi17EEEENSP_IJLi12ELi14ELi16ELi18EEEENSP_IJLi19EEEENSP_IJLi20EEEENSP_IJLi22EEEENSP_IJLi21EEEEEEENS2_IJNSP_IJLi1ELi2ELi3ELi4ELi5EEEESW_SX_SY_SZ_S10_NSP_IJLi11EEEENSP_IJLi12ELi13EEEENSP_IJLi14ELi15EEEENSP_IJLi16ELi17EEEENSP_IJLi18EEEES13_S14_S16_S15_NSP_IJLi23ELi24EEEENSP_IJLi25EEEEEEENSP_IJLi23ELi25ELi24EEEElEENS4_INS2_IJNS5_INS2_IJiiiEEENS2_IJiiS8_EEELb0EEESC_NSH_ISF_EESL_SL_SN_SC_EEENS2_IJSQ_SR_NSP_IJLi2ELi3EEEESU_SV_SX_SW_EEENS2_IJNSP_IJLi1ELi2ELi3EEEESU_SV_SW_SX_NSP_IJLi8ELi9EEEES10_EEENSP_IJLi8ELi10ELi9EEEElEENS2_IJNS4_INS2_IJNS5_ISF_NS2_IJNS7_IiLi0EEES8_EEELb0EEESL_SL_EEENS2_IJSQ_SR_SS_EEENS2_IJNSP_IJLi1ELi2EEEEST_SU_EEENSP_IJLi3ELi4EEEElEENS4_INS2_IJNS5_ISF_NS2_IJiS8_EEELb0EEESL_SL_EEES1Z_S21_S22_lEEEEES27_NS_16tensor_operation12element_wise11PassThroughES2B_NS2A_10AddReluAddELNS_25InMemoryDataOperationEnumE0ELi128ELi128ELi4ELi16ELi16ELi8ELi4ELi2ELi256ENSP_IJLi4ELi64ELi1EEEENSP_IJLi1ELi0ELi2EEEES2F_Li2ELi8ELi8ELb0ELb1ES2E_S2F_S2F_Li2ELi8ELi8ELb0ELb1ELi4ELi2ENSP_IJLi1ELi32ELi1ELi8EEEELi8ELi1ELNS_13LoopSchedulerE0ELNS_15PipelineVersionE0EEEDF16_DF16_NS2_IJPKDF16_S2L_EEEDF16_S2B_S2B_S2C_S1I_S1U_NS2_IJNS4_INS2_IJS1X_SL_SL_NSM_INS2_IJiNS7_IiLi128EEEEEELb0EEES2P_EEENS2_IJSQ_SR_SS_ST_SU_EEENS2_IJS20_ST_SU_NSP_IJLi5ELi6EEEENSP_IJLi7ELi8EEEEEEENSP_IJLi5ELi6ELi7ELi8EEEElEENS4_INS2_IJS25_SL_SL_S2P_S2P_EEES2R_S2U_S2V_lEEEEES2Y_NS_31BlockToCTileMap_M00_N0_M01AdaptILi128ELi128ES27_EENS29_6device12_GLOBAL__N_130ComputePtrOffsetOfStridedBatchILi2EEELb1EEEvPKT0_PKT1_T2_PT3_T4_T5_T6_iT7_T8_T9_T10_T11_T12_
:3:devprogram.cpp           :2979: 15459456546 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: _ZN2ck48kernel_grouped_conv_fwd_multiple_d_wmma_cshuffleINS_50GridwiseGemmMultipleD_k0mk1_k0nk1_mn_wmma_cshuffleIDF16_DF16_fDF16_NS_5TupleIJDF16_DF16_EEEDF16_NS_16TensorDescriptorINS2_IJNS_5EmbedINS2_IJiiiEEENS2_IJiiNS_17integral_constantIiLi1EEEEEELb0EEENS_11PassThroughIiEENS_3PadIiiiLb0EEESC_SC_NS5_INS2_IJiiEEESF_Lb0EEESC_NS_23Merge_v2_magic_divisionISF_EESI_NS_8RightPadIiiLb0EEESK_NS_7UnMergeISF_Lb0EEESC_EEENS2_IJNS_8SequenceIJLi0EEEENSO_IJLi1EEEENSO_IJLi2EEEENSO_IJLi3EEEENSO_IJLi4EEEENSO_IJLi5EEEENSO_IJLi6EEEENSO_IJLi7ELi9EEEENSO_IJLi8ELi10EEEENSO_IJLi11EEEENSO_IJLi12EEEENSO_IJLi14EEEENSO_IJLi13EEEEEEENS2_IJNSO_IJLi1ELi2ELi3EEEEST_SU_SV_NSO_IJLi7EEEENSO_IJLi8ELi9EEEENSO_IJLi10EEEESY_SZ_S11_S10_NSO_IJLi15ELi16EEEENSO_IJLi17EEEEEEENSO_IJLi15ELi17ELi16EEEElEENS4_INS2_IJSA_SC_SI_SK_SK_SM_SC_EEENS2_IJSP_SQ_NSO_IJLi2ELi3EEEEST_SU_S14_SV_EEENS2_IJS13_ST_SU_SV_S14_S15_S16_EEENSO_IJLi8ELi10ELi9EEEElEENS2_IJNS4_INS2_IJNS5_ISF_NS2_IJNS7_IiLi0EEES8_EEELb0EEESK_SK_EEENS2_IJSP_SQ_SR_EEENS2_IJNSO_IJLi1ELi2EEEESS_ST_EEENSO_IJLi3ELi4EEEElEENS4_INS2_IJNS5_ISF_NS2_IJiS8_EEELb0EEESK_SK_EEES1M_S1O_S1P_lEEEEES1U_NS_16tensor_operation12element_wise11PassThroughES1Y_NS1X_10AddReluAddELNS_25InMemoryDataOperationEnumE0ELi128ELi128ELi4ELi16ELi16ELi8ELi4ELi2ELi256ENSO_IJLi4ELi64ELi1EEEENSO_IJLi1ELi0ELi2EEEES22_Li2ELi8ELi8ELb0ELb1ES21_S22_S22_Li2ELi8ELi8ELb0ELb1ELi4ELi2ENSO_IJLi1ELi32ELi1ELi8EEEELi8ELi1ELNS_13LoopSchedulerE0ELNS_15PipelineVersionE0EEEDF16_DF16_NS2_IJPKDF16_S28_EEEDF16_S1Y_S1Y_S1Z_S1B_S1H_NS2_IJNS4_INS2_IJS1K_SK_SK_NSL_INS2_IJiNS7_IiLi128EEEEEELb0EEES2C_EEENS2_IJSP_SQ_SR_SS_ST_EEENS2_IJS1N_SS_ST_NSO_IJLi5ELi6EEEENSO_IJLi7ELi8EEEEEEENSO_IJLi5ELi6ELi7ELi8EEEElEENS4_INS2_IJS1S_SK_SK_S2C_S2C_EEES2E_S2H_S2I_lEEEEES2L_NS_31BlockToCTileMap_M00_N0_M01AdaptILi128ELi128ES1U_EENS1W_6device12_GLOBAL__N_130ComputePtrOffsetOfStridedBatchILi2EEELb1EEEvPKT0_PKT1_T2_PT3_T4_T5_T6_iT7_T8_T9_T10_T11_T12_
:3:devprogram.cpp           :2979: 15459456792 us: 231610: [tid:0x7f29846703c0] For Init/Fini: Kernel Name: _ZN2ck48kernel_grouped_conv_fwd_multiple_d_wmma_cshuffleINS_50GridwiseGemmMultipleD_k0mk1_k0nk1_mn_wmma_cshuffleIDF16_DF16_fDF16_NS_5TupleIJDF16_DF16_EEEDF16_NS_16TensorDescriptorINS2_IJNS_5EmbedINS2_IJiiiEEENS2_IJiiNS_17integral_constantIiLi1EEEEEELb0EEENS_11PassThroughIiEENS_3PadIiiiLb0EEESC_SC_NS5_INS2_IJiiEEESF_Lb0EEESC_NS_23Merge_v2_magic_divisionISF_EESI_NS_8RightPadIiiLb0EEESK_NS_7UnMergeISF_Lb0EEESC_EEENS2_IJNS_8SequenceIJLi0EEEENSO_IJLi1EEEENSO_IJLi2EEEENSO_IJLi3EEEENSO_IJLi4EEEENSO_IJLi5EEEENSO_IJLi6EEEENSO_IJLi7ELi9EEEENSO_IJLi8ELi10EEEENSO_IJLi11EEEENSO_IJLi12EEEENSO_IJLi14EEEENSO_IJLi13EEEEEEENS2_IJNSO_IJLi1ELi2ELi3EEEEST_SU_SV_NSO_IJLi7EEEENSO_IJLi8ELi9EEEENSO_IJLi10EEEESY_SZ_S11_S10_NSO_IJLi15ELi16EEEENSO_IJLi17EEEEEEENSO_IJLi15ELi17ELi16EEEElEENS4_INS2_IJSA_SC_SI_SK_SK_SM_SC_EEENS2_IJSP_SQ_NSO_IJLi2ELi3EEEEST_SU_S14_SV_EEENS2_IJS13_ST_SU_SV_S14_S15_S16_EEENSO_IJLi8ELi10ELi9EEEElEENS2_IJNS4_INS2_IJNS5_ISF_NS2_IJNS7_IiLi0EEES8_EEELb0EEESK_SK_EEENS2_IJSP_SQ_SR_EEENS2_IJNSO_IJLi1ELi2EEEESS_ST_EEENSO_IJLi3ELi4EEEElEENS4_INS2_IJNS5_ISF_NS2_IJiS8_EEELb0EEESK_SK_EEES1M_S1O_S1P_lEEEEES1U_NS_16tensor_operation12element_wise11PassThroughES1Y_NS1X_10AddReluAddELNS_25InMemoryDataOperationEnumE0ELi128ELi128ELi4ELi16ELi16ELi8ELi4ELi2ELi256ENSO_IJLi4ELi64ELi1EEEENSO_IJLi1ELi0ELi2EEEES22_Li2ELi8ELi8ELb0ELb1ES21_S22_S22_Li2ELi8ELi8ELb0ELb1ELi4ELi2ENSO_IJLi1ELi32ELi1ELi8EEEELi8ELi1ELNS_13LoopSchedulerE0ELNS_15PipelineVersionE0EEEDF16_DF16_NS2_IJPKDF16_S28_EEEDF16_S1Y_S1Y_S1Z_S1B_S1H_NS2_IJNS4_INS2_IJS1K_SK_SK_NSL_INS2_IJiNS7_IiLi128EEEEEELb0EEES2C_EEENS2_IJSP_SQ_SR_SS_ST_EEENS2_IJS1N_SS_ST_NSO_IJLi5ELi6EEEENSO_IJLi7ELi8EEEEEEENSO_IJLi5ELi6ELi7ELi8EEEElEENS4_INS2_IJS1S_SK_SK_S2C_S2C_EEES2E_S2H_S2I_lEEEEES2L_NS_31BlockToCTileMap_M00_N0_M01AdaptILi128ELi128ES1U_EENS1W_6device12_GLOBAL__N_130ComputePtrOffsetOfStridedBatchILi2EEELb0EEEvPKT0_PKT1_T2_PT3_T4_T5_T6_iT7_T8_T9_T10_T11_T12_
:4:command.cpp              :349 : 15459456804 us: 231610: [tid:0x7f29846703c0] Command (KernelExecution) enqueued: 0x1ddb7f0
:3:rocvirtual.cpp           :703 : 15459456807 us: 231610: [tid:0x7f29846703c0] Arg0:   = ptr:0x7f2854600000 obj:[0x7f2854600000-0x7f285bc26000]
:3:rocvirtual.cpp           :703 : 15459456809 us: 231610: [tid:0x7f29846703c0] Arg1:   = ptr:0x7f2970c00000 obj:[0x7f2970c00000-0x7f2972700000]
:3:rocvirtual.cpp           :778 : 15459456810 us: 231610: [tid:0x7f29846703c0] Arg2:   = val:139815957823488
:3:rocvirtual.cpp           :703 : 15459456811 us: 231610: [tid:0x7f29846703c0] Arg3:   = ptr:0x7f2861000000 obj:[0x7f2861000000-0x7f2863880000]
:3:rocvirtual.cpp           :778 : 15459456812 us: 231610: [tid:0x7f29846703c0] Arg4:   = val:137453070468
:3:rocvirtual.cpp           :778 : 15459456813 us: 231610: [tid:0x7f29846703c0] Arg5:   = val:144115188612781928
:3:rocvirtual.cpp           :778 : 15459456814 us: 231610: [tid:0x7f29846703c0] Arg6:   = val:562949955518679
:3:rocvirtual.cpp           :778 : 15459456815 us: 231610: [tid:0x7f29846703c0] Arg7:   = val:8589934624
:3:rocvirtual.cpp           :778 : 15459456816 us: 231610: [tid:0x7f29846703c0] Arg8:   = val:304942678018
:3:rocvirtual.cpp           :778 : 15459456817 us: 231610: [tid:0x7f29846703c0] Arg9:   = val:38654705920
:3:rocvirtual.cpp           :778 : 15459456818 us: 231610: [tid:0x7f29846703c0] Arg10:   = val:1099511630368
:3:rocvirtual.cpp           :778 : 15459456819 us: 231610: [tid:0x7f29846703c0] Arg11:   = val:1099511630368
:3:rocvirtual.cpp           :778 : 15459456820 us: 231610: [tid:0x7f29846703c0] Arg12:   = val:-9223372036854775800
:3:rocvirtual.cpp           :778 : 15459456821 us: 231610: [tid:0x7f29846703c0] Arg13:   = val:1899956092797120
:3:rocvirtual.cpp           :2774: 15459456822 us: 231610: [tid:0x7f29846703c0] ShaderName : _ZN2ck48kernel_grouped_conv_fwd_multiple_d_wmma_cshuffleINS_50GridwiseGemmMultipleD_k0mk1_k0nk1_mn_wmma_cshuffleIDF16_DF16_fDF16_NS_5TupleIJDF16_DF16_EEEDF16_NS_16TensorDescriptorINS2_IJNS_5EmbedINS2_IJiiiiEEENS2_IJiiiNS_17integral_constantIiLi1EEEEEELb0EEENS_11PassThroughIiEENS_3PadIiiiLb0EEESE_SC_SC_NS5_INS2_IJiiEEESF_Lb0EEESG_SC_NS_23Merge_v2_magic_divisionINS2_IJiiiEEEEESJ_NS_8RightPadIiiLb0EEESL_NS_7UnMergeISF_Lb0EEESC_EEENS2_IJNS_8SequenceIJLi0EEEENSP_IJLi1EEEENSP_IJLi2EEEENSP_IJLi3EEEENSP_IJLi4EEEENSP_IJLi5EEEENSP_IJLi6EEEENSP_IJLi7EEEENSP_IJLi8EEEENSP_IJLi9ELi11ELi13EEEENSP_IJLi10ELi12ELi14EEEENSP_IJLi15EEEENSP_IJLi16EEEENSP_IJLi18EEEENSP_IJLi17EEEEEEENS2_IJNSP_IJLi1ELi2ELi3ELi4EEEESV_SW_SX_SY_NSP_IJLi9EEEENSP_IJLi10ELi11EEEENSP_IJLi12ELi13EEEENSP_IJLi14EEEES11_S12_S14_S13_NSP_IJLi19ELi20EEEENSP_IJLi21EEEEEEENSP_IJLi19ELi21ELi20EEEElEENS4_INS2_IJNS5_ISI_NS2_IJiiS8_EEELb0EEESC_NSH_ISF_EESL_SL_SN_SC_EEENS2_IJSQ_SR_NSP_IJLi2ELi3EEEESU_SV_SX_SW_EEENS2_IJNSP_IJLi1ELi2ELi3EEEESU_SV_SW_SX_NSP_IJLi8ELi9EEEENSP_IJLi10EEEEEEENSP_IJLi8ELi10ELi9EEEElEENS2_IJNS4_INS2_IJNS5_ISF_NS2_IJNS7_IiLi0EEES8_EEELb0EEESL_SL_EEENS2_IJSQ_SR_SS_EEENS2_IJNSP_IJLi1ELi2EEEEST_SU_EEENSP_IJLi3ELi4EEEElEENS4_INS2_IJNS5_ISF_NS2_IJiS8_EEELb0EEESL_SL_EEES1W_S1Y_S1Z_lEEEEES24_NS_16tensor_operation12element_wise11PassThroughES28_NS27_10AddReluAddELNS_25InMemoryDataOperationEnumE0ELi128ELi128ELi4ELi16ELi16ELi8ELi4ELi2ELi256ENSP_IJLi4ELi64ELi1EEEENSP_IJLi1ELi0ELi2EEEES2C_Li2ELi8ELi8ELb0ELb1ES2B_S2C_S2C_Li2ELi8ELi8ELb0ELb1ELi4ELi2ENSP_IJLi1ELi32ELi1ELi8EEEELi8ELi1ELNS_13LoopSchedulerE0ELNS_15PipelineVersionE0EEEDF16_DF16_NS2_IJPKDF16_S2I_EEEDF16_S28_S28_S29_S1F_S1R_NS2_IJNS4_INS2_IJS1U_SL_SL_NSM_INS2_IJiNS7_IiLi128EEEEEELb0EEES2M_EEENS2_IJSQ_SR_SS_ST_SU_EEENS2_IJS1X_ST_SU_NSP_IJLi5ELi6EEEENSP_IJLi7ELi8EEEEEEENSP_IJLi5ELi6ELi7ELi8EEEElEENS4_INS2_IJS22_SL_SL_S2M_S2M_EEES2O_S2R_S2S_lEEEEES2V_NS_31BlockToCTileMap_M00_N0_M01AdaptILi128ELi128ES24_EENS26_6device12_GLOBAL__N_130ComputePtrOffsetOfStridedBatchILi2EEELb1EEEvPKT0_PKT1_T2_PT3_T4_T5_T6_iT7_T8_T9_T10_T11_T12_
:4:rocvirtual.cpp           :862 : 15459456830 us: 231610: [tid:0x7f29846703c0] HWq=0x7f2984600000, Dispatch Header = 0x1502 (type=2, barrier=1, acquire=2, release=2), setup=3, grid=[344064, 1, 1], workgroup=[256, 1, 1], private_seg_size=0, group_seg_size=32768, kernel_obj=0x7f2982bdd980, kernarg_address=0x7f2970400000, completion_signal=0x0
:3:hip_module.cpp           :470 : 15459456833 us: 231610: [tid:0x7f29846703c0] hipLaunchKernel: Returned hipSuccess :
:3:hip_event.cpp            :305 : 15459456835 us: 231610: [tid:0x7f29846703c0]  hipEventCreate ( 0x7ffcada24cc0 )
:3:hip_event.cpp            :311 : 15459456838 us: 231610: [tid:0x7f29846703c0] hipEventCreate: Returned hipSuccess : event:0x1e54960
:3:hip_event.cpp            :305 : 15459456839 us: 231610: [tid:0x7f29846703c0]  hipEventCreate ( 0x7ffcada24cb8 )
:3:hip_event.cpp            :311 : 15459456840 us: 231610: [tid:0x7f29846703c0] hipEventCreate: Returned hipSuccess : event:0x1e82ac0
:3:hip_device_runtime.cpp   :502 : 15459456842 us: 231610: [tid:0x7f29846703c0]  hipDeviceSynchronize (  )
:3:rocdevice.cpp            :2645: 15459456843 us: 231610: [tid:0x7f29846703c0] No HW event
:4:commandqueue.cpp         :135 : 15459456844 us: 231610: [tid:0x7f29846703c0] HW Event not ready, awaiting completion instead
:4:command.cpp              :289 : 15459456846 us: 231610: [tid:0x7f29846703c0] Queue marker to command queue: 0x1d48940:4:command.cpp              :349 : 15459456847 us: 231610: [tid:0x7f29846703c0] Command (InternalMarker) enqueued: 0x21171a0
:4:rocvirtual.cpp           :1005: 15459456849 us: 231610: [tid:0x7f29846703c0] HWq=0x7f2984600000, BarrierAND Header = 0x1503 (type=3, barrier=1, acquire=2, release=2), dep_signal=[0x0, 0x0, 0x0, 0x0, 0x0], completion_signal=0x7f29847ee480:4:rocvirtual.cpp           :550 : 15459456850 us: 231610: [tid:0x7f29846703c0] Host wait on completion_signal=0x7f29847ee480
:3:rocvirtual.hpp           :61  : 15459456851 us: 231610: [tid:0x7f29846703c0] Host active wait for Signal = (0x7f29847ee480) for -1 ns
Memory access fault by GPU node-1 (Agent handle: 0x1d584f0) on address 0x7f295458e000. Reason: Page not present or supervisor privilege.
Aborted (core dumped)
```

**Comment by aska-0096** [comment]
A guess based on the log info from 7900xt (provided above) and 7900xtx (on my develop machine) point to it is a driver related issue

```
:3:rocvirtual.cpp           :703 : 15459456807 us: 231610: [tid:0x7f29846703c0] Arg0:   = ptr:0x7f2854600000 obj:[0x7f2854600000-0x7f285bc26000]
:3:rocvirtual.cpp           :703 : 15459456809 us: 231610: [tid:0x7f29846703c0] Arg1:   = ptr:0x7f2970c00000 obj:[0x7f2970c00000-0x7f2972700000]
:3:rocvirtual.cpp           :778 : 15459456810 us: 231610: [tid:0x7f29846703c0] Arg2:   = val:139815957823488
:3:rocvirtual.cpp           :703 : 15459456811 us: 231610: [tid:0x7f29846703c0] Arg3:   = ptr:0x7f2861000000 obj:[0x7f2861000000-0x7f2863880000]
:3:rocvirtual.cpp           :778 : 15459456812 us: 231610: [tid:0x7f29846703c0] Arg4:   = val:137453070468
:3:rocvirtual.cpp           :778 : 15459456813 us: 231610: [tid:0x7f29846703c0] Arg5:   = val:144115188612781928
:3:rocvirtual.cpp           :778 : 15459456814 us: 231610: [tid:0x7f29846703c0] Arg6:   = val:562949955518679
:3:rocvirtual.cpp           :778 : 15459456815 us: 231610: [tid:0x7f29846703c0] Arg7:   = val:8589934624
:3:rocvirtual.cpp           :778 : 15459456816 us: 231610: [tid:0x7f29846703c0] Arg8:   = val:304942678018
:3:rocvirtual.cpp           :778 : 15459456817 us: 231610: [tid:0x7f29846703c0] Arg9:   = val:38654705920
:3:rocvirtual.cpp           :778 : 15459456818 us: 231610: [tid:0x7f29846703c0] Arg10:   = val:1099511630368
:3:rocvirtual.cpp           :778 : 15459456819 us: 231610: [tid:0x7f29846703c0] Arg11:   = val:1099511630368

**:3:rocvirtual.cpp           :778 : 15459456820 us: 231610: [tid:0x7f29846703c0] Arg12:   = val:-9223372036854775800**

:3:rocvirtual.cpp           :778 : 15459456821 us: 231610: [tid:0x7f29846703c0] Arg13:   = val:1899956092797120
```

The Arg 12 here is weird, in XTX its value is: 8
Arg12 is to describe how workgroup mapped on whole problem size, with the incorrect workgroup mapping information, mem access fault could be triggered easily.

The relationship between 8 and  -9223372036854775800 is they shared same hex16 0x8000000000000008.

@atafra a suggestion from me is, update your amdgpu driver if it is not the latest one. although I can not grant kernel will executed without mem access fault, we have less chance to meet this kind of issue.

**Comment by zjing14** [comment]
@atafra Another suggestion is trying to replace the flag "-O2" with "-O0" or "-O1".

**Comment by atafra** [comment]
@zjing14 Compiling with "-O0" does not work, I get the following compile error:
```
error: Illegal instruction detected: Operand has incorrect register class.
renamable $vgpr0_vgpr1_vgpr2_vgpr3 = BUFFER_LOAD_DWORDX4_OFFEN killed renamable $vgpr0, killed renamable $sgpr4_sgpr5_sgpr6_sgpr7, killed renamable $vgpr1, 0, 0, 0, 0, implicit $exec :: (dereferenceable load (s128), align 1, addrspace 4)
error: Illegal instruction detected: Operand has incorrect register class.
BUFFER_STORE_DWORDX4_OFFEN_exact killed renamable $vgpr0_vgpr1_vgpr2_vgpr3, killed renamable $vgpr4, killed renamable $sgpr4_sgpr5_sgpr6_sgpr7, killed renamable $vgpr5, 0, 0, 0, 0, implicit $exec :: (dereferenceable store (s128), align 1, addrspace 4)
2 errors generated when compiling for gfx1100.
```

Compiling with "-O1" works, but I get the same memory access fault.

I'm already using the latest public driver which ships with ROCm 5.4.3. Could you validate your code with this? If the code works only with some internal driver or compiler, nobody outside AMD could use it.

Is there any chance you could validate on a 7900 XT? The problem is that if this issue won't be fixed within the next couple of weeks, we will have to ship our product without RDNA 3 support.

**Comment by zjing14** [comment]
@atafra We've purchased a 7900xt. We should get it early next week and start to debug the issue.

**Comment by zjing14** [comment]
@atafra Update: We have set up the 7900xt machine, and can reproduce the issue.

**Comment by aska-0096** [comment]
Hi @atafra 
It appears that the root cause is a slight change in the market product and engineering samples. The temporary solution is:

Please change the following two flag to 0
https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/f80776d937b9009654a7cb6e1d3f9584cd7b6e41/include/ck/ck.hpp#L71-L75

Turn these two options off may induce performance drop, I'm finding a solution to get it back with correctness.

**Comment by atafra** [comment]
I can confirm that this temporary solution fixes the crash, and the results are correct as well. Thanks!

**Comment by aska-0096** [comment]
Hi @atafra , a fix provided through PR #659, let me know is it work for you.

**Comment by atafra** [comment]
@aska-0096 The fix in the PR works, thanks! No crash, output is correct, performance is slightly better than with temporary fix.

**Comment by zjing14** [comment]
@atafra @aska-0096 Great, thank you both. Let us close the issue.
